src/Compilers/VisualBasic/Portable/Binding/InitializerSemanticModel.vb

```diff
@@ -133,7 +133,8 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                     Throw ExceptionUtilities.UnexpectedValue(Me.MemberSymbol.Kind)
             End Select
 
-            Return boundInitializer
+            Dim expressionInitializer = TryCast(boundInitializer, BoundExpression)
+            Return If(expressionInitializer IsNot Nothing AndAlso expressionInitializer.Kind <> BoundKind.BadExpression, New BoundEqualsValue(initializer, expressionInitializer), boundInitializer)
```

Comment:
The issue turned out to be that flow analysis was looking for a BoundExpression and instead found a BoundEqualsValue. Going through the BoundEqualsValue to a BoundExpression solved the problem.

However, I have no idea how many other lurking issues of this sort might remain. The more I think about it, the more I'm worried that introducing a placeholder node that any client must recognize and skip through is not the best behavior. Mapping multiple syntax nodes to one bound node in a node map may feel a little unclean, but it completely avoids the possibility of introducing problems like this.

Also, VB has already been doing something much like this. I found this code in MemberSemanticModel.vb:

If Not _guardedNodeMap.ContainsKey(bindingRoot) Then
                ' Generally 'bindingRoot' is supposed to be found in node map at this point,
                ' but it will not happen in some scenarios such as for field or property 
                ' initializers, let's add it to prevent re-binding 

                Debug.Assert(bindingRoot.Kind = SyntaxKind.FieldDeclaration OrElse
                             bindingRoot.Kind = SyntaxKind.PropertyStatement OrElse
                             bindingRoot.Kind = SyntaxKind.Parameter OrElse
                             bindingRoot.Kind = SyntaxKind.EnumMemberDeclaration OrElse
                             bindingRoot Is Me.Root AndAlso Me.IsSpeculativeSemanticModel)

                _guardedNodeMap.Add(bindingRoot, ImmutableArray.Create(Of BoundNode)(boundRoot))